### PR TITLE
Feat: (registerFaustBlock) Add Save Function

### PR DESCRIPTION
**Pull request by theodesp**
## Tasks

- [ ] I have signed a [Contributor License Agreement (CLA)](https://github.com/wpengine/faustjs#contributor-license-agreement) with WP Engine.
- [ ] If a code change, I have written testing instructions that the whole team & outside contributors can understand.
- [ ] I have written and included a comprehensive [changeset](https://github.com/wpengine/faustjs/blob/canary/DEVELOPMENT.md#deployment) to properly document the changes I've made.

## Description

* Adds default Save function when using `registerFaustBlock`
* Unit tests.

## Related Issue(s):

https://github.com/wpengine/faustjs/issues/1522

## Testing

1. run `npm install`
2. Build the project 
3. cd inside the project folder:
   `cd packages/block-editor-utils`
5. run `npm link`
6. On your WordPress instance create a new block plugin using [@wordpress/create-block](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/)
7. Inside the plugin folder run `npm link @faustwp/block-editor-utils ` to add a reference to the package.
8. Rewrite the genenated `index.js` editor script to use the `registerFaustBlock` helper function instead. 
```js
// index.js
import './style.scss';
// import block.json
import metadata from './block.json';

// import Pure React Component
import MyFirstBlock from './MyFirstBlock';

import Edit from './edit';

// Register React Component in Gutenberg
import registerFaustBlock from '@faustwp/block-editor';

registerFaustBlock(MyFirstBlock, {blockJson: metadata, editFn: (ctx) => Edit(ctx.props) })
```
Here `MyFirstBlock` is a simple Div Component:

```
// MyFirstBlock.js
function MyFirstBlock({ style, className, attributes, children, ...props }) {
	const styles = {
		...style,
		backgroundColor: attributes.bg_color,
		color: attributes.text_color,
	};
	return (
		<div
			{...props}
			style={styles}
			className={className}
			dangerouslySetInnerHTML={{ __html: attributes.message }}
		/>
	);
}
```

```json
// block.json
{
	"$schema": "https://schemas.wp.org/trunk/block.json",
	"apiVersion": 2,
	"name": "create-block/my-first-block",
	"version": "0.1.0",
	"title": "My First Block",
	"category": "widgets",
	"icon": "smiley",
	"description": "My block",
	"supports": {
		"html": false
	},
	"attributes": {
		"message": {
		   "type": "string",
		   "source": "text",
                    "selector": "div",
                    "default": ""
		},
		"bg_color": { "type": "string", "default": "#000000" },
        "text_color": { "type": "string", "default": "#ffffff" }
	},
	"textdomain": "my-first-block",
	"editorScript": "file:./index.js",
	"editorStyle": "file:./index.css",
	"style": "file:./style-index.css"
}
```

7) Now go into the editor and try to create a new block. It should save the block component static markup in the database as usual.

### CODE REVIEW

**Comment by josephfusco** [comment]
Hey @theodesp in regards to the testing instructions, I ran `npm link` inside of `packages/block-editor-utils`.

I am able to see the component show up in the editor following your instructions. I will note that the edit functionality is not working but that seems expected given that MERL-1142 is in progress.

Screenshot 2023-08-18 at 10 50 37 AM

**Review by josephfusco — APPROVED** [review]

**Review by blakewilson — CHANGES_REQUESTED** [review]
Nice @theodesp! Ran through the testing steps and all seemed to work fine. Left two questions related to the code.

File: packages/block-editor-utils/package.json [hunk 1]
```diff
@@ -25,10 +26,12 @@
     "rimraf": "^4.4.0",
     "@wordpress/jest-preset-default": "^11.9.0",
     "@types/wordpress__blocks": "12.5.0",
+    "@types/wordpress__block-editor": "11.5.1",
     "ts-jest": "29.1.1",
     "jest": "29.6.2",
     "react": ">=18.0.0",
-    "react-dom": ">=18.0.0"
+    "react-dom": ">=18.0.0",
+    "@react-spring/web": "9.7.3"
```
[the remark this answers is not in the document]
Comment (theodesp) [reply packages/block-editor-utils/package.json:34]: I had to include this for running the test cases for some reason I got module resolution errors. I will try to remove it again and see if that works.

File: packages/block-editor-utils/src/components/Save.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { useBlockProps, InnerBlocks } from '@wordpress/block-editor';
+import { SaveFnContext } from '../registerFaustBlock.js';
+
+export default function Save<T extends Record<string, any>>(
+  ctx: SaveFnContext<T>,
+) {
+  const blockProps = useBlockProps.save();
+  const { block: Block, props } = ctx;
+  const allBlockProps: React.Attributes = {
+    key: null,
```
[the remark this answers is not in the document]
Comment (theodesp) [reply packages/block-editor-utils/src/components/Save.tsx:11]: `key` was needed for Typescript to compile. By default the key should be left `null`

**Comment by theodesp** [comment]
@josephfusco 
The editFn accepts a ctx object and if you look at the example I mention you need to pass the `ctx.props` to the Edit function.

> ```js
> editFn: (ctx) => Edit(ctx.props)
> ```

File: packages/block-editor-utils/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (blakewilson) [reply packages/block-editor-utils/package.json:34]: @theodesp Gotcha. It is a dev dependency so not that big of a deal. Just wanted to make sure.

**Review by blakewilson — APPROVED** [review]

**Pull request merged**